hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/UpdateProcessorTest.java

```diff
@@ -108,6 +120,56 @@ public void when_keyDoesNotExist_then_doesNotCreateEntry() {
         assertThat(map).containsExactly(entry(1, 1));
     }
 
+    @Test
+    public void writeAndReadTest_success() throws IOException {
+        final var evalContextWithSubjectMock = mock(ExpressionEvalContext.class);
+        when(evalContextWithSubjectMock.getSerializationService()).thenReturn(mock());
+        final var subject = new Subject(true, singleton(mock(Principal.class)), emptySet(), emptySet());
+        when(evalContextWithSubjectMock.subject()).thenReturn(subject);
```

Comment:
you mock objects very deeply. This thing is not easy to unit test purely because of many dependent object. IMO better and simpler would be to create integration test. HZ instance starts very quickly (it is already started in `UpdateProcessorTest`). If you get `SerializationService` from it, then the test will be trivial using toData/toObject.